synapse/handlers/auth.py

```diff
@@ -865,7 +866,7 @@ def validate_hash(self, password, stored_hash):
         def _do_validate_hash():
             return bcrypt.checkpw(
                 password.encode('utf8') + self.hs.config.password_pepper,
-                stored_hash.encode('utf8')
+                stored_hash.encode('ascii')
```

Comment:
I'm curious about this too.

Reply:
The hash should only ever be 7-bit ASCII, according to https://passlib.readthedocs.io/en/stable/modular_crypt_format.html, so it's just a case of being strict in what we accept, here.